test/cases/regresssionTests/1427/original.ts

```diff
@@ -0,0 +1,80 @@
+import ts from "typescript";
+
+(function () {
+	interface FunctionCallType {
+		parameters?: (ts.Type | undefined)[];
+		returnValue?: ts.Type;
+	}
+	interface CombinedFunctionType {
+		parameters: ts.Type[][];
+		returnValue?: ts.Type[];
+	}
+
+	const functionCallTypes: FunctionCallType[] = [];
+	const combinedType = functionCallTypes.reduce<CombinedFunctionType>(
+		(accumulator, functionCallType) => {
+			return {
+				parameters: combineParameters(
+					undefined,
+					accumulator.parameters,
+					functionCallType.parameters,
+				),
+				returnValue: collectOptionals(accumulator.returnValue, [
+					functionCallType.returnValue,
+				]).filter(isNotUndefined),
+			};
+		},
+		{
+			parameters: [],
+			returnValue: [],
+		},
+	);
+
+	const combineParameters = (
+		request: unknown,
+		previous: ts.Type[][],
+		next: (ts.Type | undefined)[] | undefined,
+	) => {
+		if (next === undefined) {
+			return previous;
+		}
+
+		const combined: ts.Type[][] = [];
+		let i: number;
+
+		for (i = 0; i < previous.length; i += 1) {
+			combined.push([...previous[i]]);
+
+			if (i < next.length && next[i] !== undefined) {
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				combined[i].push(next[i]!);
+			}
+		}
+
+		for (i; i < next.length; i += 1) {
+			if (next[i] !== undefined) {
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				combined.push([next[i]!]);
+			}
+		}
+
+		return combined;
+	};
+
+	const collectOptionals = <T>(
+		...arrays: (readonly T[] | undefined)[]
+	): T[] => {
+		const results: T[] = [];
+
+		for (const array of arrays) {
+			if (array !== undefined) {
+				results.push(...array);
+			}
+		}
+
+		return results;
+	};
+
+	const isNotUndefined = <T>(item: T | undefined): item is T =>
+		item !== undefined;
+})();
```

Comment:
> not really test about how the fixes work otherwise

I think it is 😄. It's just testing an edge case or two that wasn't accounted for previously. A lot of the tests were added in as "regression" tests, but are now named generally.

I filed #1504 just now, too, for splitting up tests.

For this one, we can reproduce the same issue with a more targeted snippet:

```suggestion
(function () {
	interface Shape {
		value?: unknown;
	}

	[].reduce<Shape>(() => ({}), {});
})();
```

...which makes me think this could be named, maybe... `test/cases/fixes/incompleteTypes/nonGenericInterfaceAsTypeArgument` or something like that?

Was there anything else you'd wanted to test in here too?